src/geom/elem.C

```diff
@@ -399,16 +402,16 @@ dof_id_type Elem::key () const
 {
   const unsigned short n_n = this->n_nodes();
 
-  std::vector<dof_id_type> node_ids(n_n);
+  std::array<dof_id_type, Elem::max_n_nodes> node_ids;
 
   for (unsigned short n=0; n != n_n; ++n)
     node_ids[n] = this->node_id(n);
 
   // Always sort, so that different local node numberings hash to the
   // same value.
-  std::sort (node_ids.begin(), node_ids.end());
+  std::sort (node_ids.begin(), node_ids.begin()+n_n);
 
-  return Utility::hashword(node_ids);
+  return Utility::hashword(node_ids.begin(), n_n);
```

Comment:
So this is calling one of the overloads
```
uint32_t hashword(const uint32_t * k, size_t length, uint32_t initval=0)
uint64_t hashword(const uint64_t * k, size_t length);
```
depending on `dof_id_type`. Is it guaranteed that `std::array<T>::begin()` returns `T*`?

The docs [here](https://en.cppreference.com/w/cpp/container/array/begin) and [here](https://en.cppreference.com/w/cpp/container/array) refer to an `iterator` type but I don't see any guarantees of this...

Reply:
Seems to me that `node_ids.data()` is the more correct thing to use, though I expect in almost all cases they are going to be equivalent.

Reply:
Oh, hell, I am passing begin() to the hashword method.  Thanks; I'll fix that.  Bugs that work fine on most compilers until things scream and die on newer compilers are not friendly sorts of bugs.  At least this one would have been a compile-time error.  Thanks for catching it!

Reply:
The line I commented on is passing begin() to hashword, unless I need to hit refresh...

Reply:
I did remember to use data() in hashword.h so I haven't *completely* lost my mind...